syntax = "proto3";

package pairing_server;

import "types.proto";

message RegisterResultRequest {
    Identification gameId = 1;
    Result result = 2;
}

service PairingServer {
    rpc GetTournament(Identification) returns (Tournament) {}
    rpc GetPlayers(Identification) returns (stream Player) {}
    rpc GetGames(Identification) returns (stream Game) {}

    rpc CreateTournament(Tournament) returns (Identification) {}
    rpc PairNextRound(Identification) returns (stream Game) {}

    rpc SignupPlayer(Player) returns (Identification) {}
    rpc RegisterResult(RegisterResultRequest) returns (Identification) {}
}
